lms/djangoapps/grades/management/commands/reset_grades.py

```diff
@@ -0,0 +1,144 @@
+        )
+        parser.add_argument(
+            '--delete',
+            action='store_true',
+            default=False,
+            dest='delete',
+            help="Actually perform the deletions of the course. For a Dry Run, use --dry_run instead."
+        )
+        parser.add_argument(
+            '--courses',
+            dest='courses',
+            nargs='+',
+            help='Reset persistent grades for the list of courses provided.',
+        )
+        parser.add_argument(
+            '--all_courses',
+            action='store_true',
+            dest='all_courses',
+            default=False,
+            help='Reset persistent grades for all courses.',
+        )
+        parser.add_argument(
+            '--modified_start',
+            dest='modified_start',
+            help='Starting range for modified date (inclusive): e.g. "2016-08-23 16:43"',
+        )
+        parser.add_argument(
+            '--modified_end',
+            dest='modified_end',
+            help='Ending range for modified date (inclusive): e.g. "2016-12-23 16:43"',
+        )
+
+    def handle(self, *args, **options):
+        course_keys = None
+        modified_start = None
+        modified_end = None
+
+        if not options.get('delete') and not options.get('dry_run'):
+            raise CommandError('Either --delete or --dry_run must be specified.')
+
+        if not options.get('courses') and not options.get('all_courses'):
+            raise CommandError('Either --courses or --all_courses must be specified.')
+
+        if options.get('courses'):
+            try:
+                course_keys = [CourseKey.from_string(course_key_string) for course_key_string in options['courses']]
+            except InvalidKeyError:
+                raise CommandError('Invalid key specified.')
+
+        dry_run = options.get('dry_run', False)
+        log.info("reset_grade: Started in %s mode!", "dry_run" if dry_run else "real")
+
+        if options.get('modified_start'):
+            modified_start = datetime.strptime(options['modified_start'], DATE_FORMAT)
+
+        if options.get('modified_end'):
+            if not modified_start:
+                raise CommandError('Optional value for modified_end provided without a value for modified_start.')
+            modified_end = datetime.strptime(options['modified_end'], DATE_FORMAT)
+
+        if dry_run:
+            self._query_grades(PersistentSubsectionGrade, course_keys, modified_start, modified_end)
+            self._query_grades(PersistentCourseGrade, course_keys, modified_start, modified_end)
+
+        else:
+            self._delete_grades(PersistentSubsectionGrade, course_keys, modified_start, modified_end)
+            self._delete_grades(PersistentCourseGrade, course_keys, modified_start, modified_end)
+
+        log.info("reset_grade: Finished in %s mode!", "dry_run" if dry_run else "real")
```

Comment:
Nit: maybe instead of "real" mode it could be called "delete" mode to reflect the arg.